relayer-cli/src/components.rs

```diff
@@ -92,12 +94,33 @@ fn enable_ansi() -> bool {
     atty::is(atty::Stream::Stdout) && atty::is(atty::Stream::Stderr)
 }
 
-fn build_tracing_filter(log_level: String) -> String {
+/// Builds a tracing filter based on the input `log_level`.
+/// Enables tracing exclusively for the relayer crates.
+/// Returns error if the given `log_level` is invalid.
+fn build_tracing_filter(log_level: String) -> Result<EnvFilter, FrameworkError> {
     let target_crates = ["ibc_relayer", "ibc_relayer_cli"];
 
-    target_crates
+    // SAFETY: unwrap() below works as long as `target_crates` is not empty.
+    let directive_raw = target_crates
         .iter()
         .map(|&c| format!("{}={}", c, log_level))
         .reduce(|a, b| format!("{},{}", a, b))
-        .unwrap()
+        .unwrap();
+
+    // Build the filter directive
+    match EnvFilter::try_new(directive_raw.clone()) {
+        Ok(out) => Ok(out),
+        Err(e) => {
+            let our_err = config::Error::InvalidLogLevel(log_level.clone(), e.to_string());
```

Comment:
Maybe we should make our enum for the `log_level` property and let serde raise an error if the level is invalid.

Reply:
Good idea 38cc315